SimPEG/potential_fields/gravity/simulation.py

```diff
@@ -27,27 +242,132 @@ class Simulation3DIntegral(BasePFSimulation):
 
         Gradient components ("gxx", "gyy", "gzz", "gxy", "gxz", "gyz") are
         returned in Eotvos (:math:`10^{-9} s^{-2}`).
+
+    Parameters
+    ----------
+    mesh : discretize.TreeMesh or discretize.TensorMesh
+        Mesh use to run the gravity simulation.
+    survey : SimPEG.potential_fields.gravity.Survey
+        Gravity survey with information of the receivers.
+    ind_active : (n_cells) array, optional
+        Array that indicates which cells in ``mesh`` are active cells.
+    rho : array (optional)
+        Density array for the active cells in the mesh.
+    rhoMap : Mapping (optional)
+        Model mapping.
+    sensitivity_dtype : numpy.dtype, optional
+        Data type that will be used to build the sensitivity matrix.
+    store_sensitivities : str
+        Options for storing sensitivity matrix. There are 3 options
+
+        - 'ram': sensitivities are stored in the computer's RAM
+        - 'disk': sensitivities are written to a directory
+        - 'forward_only': you intend only do perform a forward simulation and
+          sensitivities do not need to be stored
+
+    sensitivity_path : str, optional
+        Path to store the sensitivity matrix if ``store_sensitivities`` is set
+        to ``"disk"``. Default to "./sensitivities".
+    engine : str, optional
+       Choose which engine should be used to run the forward model:
+       ``"geoana"`` or "``choclo``".
+    choclo_parallel : bool, optional
+        If True, the simulation will run in parallel. If False, it will
+        run in serial. If ``engine`` is not ``"choclo"`` this argument will be
+        ignored.
     """
 
     rho, rhoMap, rhoDeriv = props.Invertible("Density")
 
-    def __init__(self, mesh, rho=None, rhoMap=None, **kwargs):
+    def __init__(
+        self,
+        mesh,
+        rho=None,
+        rhoMap=None,
+        engine="geoana",
```

Comment:
Ultimately, I think the engine/parallelism should be inherited from `BasePFSimulation`. The interface (should) work the same for both gravity and magnetics eventually; all we'd need for now is to add a check that the mag simulation is using `engine == "geoana"`, and raise if not

I know it's outside the scope of this PR, but I'm strongly in favor of deprecating the multiprocessing implementation as soon as possible-- it's slow, and very memory inefficient, and with the addition of `choclo`, it makes the interface quite clunky.

Reply:
I agree. Ultimately, engine and parallelism should be handled by the parent class, since they will be used for both mag and gravity simulations. I added them to the gravity simulation for this PR just to narrow its scope to only the gravity simulation. I think we can move these bits to the `BasePFSimulation` when implementing the mag simulation with choclo.

> I know it's outside the scope of this PR, but I'm strongly in favor of deprecating the multiprocessing implementation as soon as possible-- it's slow, and very memory inefficient, and with the addition of choclo, it makes the interface quite clunky.

I think these are good reasons. For now we are going to leave the two implementations and decide to ditch the multiprocessing one or not later on. There are still some features that the choclo version is not implementing, like working with Dask to distribute the forward among cluster nodes, or storing the sensitivity to disk using compressed zarr arrays, which have higher io speed and require less disk space.